cpp/src/arrow/compute/kernels/scalar_nested.cc

```diff
@@ -87,6 +89,198 @@ Status GetListElementIndex(const ExecValue& value, T* out) {
   return Status::OK();
 }
 
+template <typename Type, typename IndexType>
+struct ListSlice {
+  using offset_type = typename Type::offset_type;
+
+  static Status Exec(KernelContext* ctx, const ExecSpan& batch, ExecResult* out) {
+    const auto opts = OptionsWrapper<ListSliceOptions>::Get(ctx);
+
+    // Invariants
+    if (!opts.stop.has_value()) {
+      // TODO: Support slicing to arbitrary end
```

Comment:
Do you plan to tackle the TODOs in this PR, or have you filed follow up tickets? (If the latter, can they be referenced inline like `TODO(ARROW-NNNNN): ...`?)

Reply:
Yes, added the JIRAs and updated comments. Thanks. :)